diff --git a/baselines/aggregate_method_statistics.py b/baselines/aggregate_method_statistics.py
index 6bb2311..f1ef9d2 100644
--- a/baselines/aggregate_method_statistics.py
+++ b/baselines/aggregate_method_statistics.py
@@ -178,7 +178,7 @@ def calculate_average_homogeinity_and_completeness_score_stats(directory_path):
             df = pd.read_csv(scores_path)
 
             # Validate required columns
-            required_cols = ['error_slices_homogeneity_score', 'underperforming_populations_completeness_score']
+            required_cols = ['error_slices_purity_score', 'underperforming_populations_completeness_score']
             if not all(col in df.columns for col in required_cols):
                 print(
                     f"Warning: {scores_path} missing required columns, skipping"
@@ -300,9 +300,9 @@ def main(directory_path, output=None, aggregated_pop_ids=None):
             directory_path
         )
 
-        result_df_homogeneity_and_v_measure_score = calculate_average_homogeinity_and_completeness_score_stats(
-            directory_path
-        )
+        # result_df_homogeneity_and_v_measure_score = calculate_average_homogeinity_and_completeness_score_stats(
+        #     directory_path
+        # )
 
         result_df_underperforming_population_frequency = calculate_average_underperforming_population_frequency_in_rep_slice(
             directory_path, aggregated_pop_ids
@@ -334,16 +334,16 @@ def main(directory_path, output=None, aggregated_pop_ids=None):
                     aggregated_df.to_csv(aggregated_output, index=False)
                     print(f"\nAggregated results saved to {aggregated_output}")
 
-        print("\nHomogeneity and V-Measure Score Statistics:")
-        print("=" * 60)
-        print(result_df_homogeneity_and_v_measure_score.to_string(index=False))
+        # print("\nHomogeneity and V-Measure Score Statistics:")
+        # print("=" * 60)
+        # print(result_df_homogeneity_and_v_measure_score.to_string(index=False))
 
         # Save to CSV if output specified
         if output:
             result_df.to_csv(output, index=False)
-            v_score_path = output.replace("precision_at_10_statistics.csv", "v_score_statistics.csv")
-            result_df_homogeneity_and_v_measure_score.to_csv(v_score_path, index=False)
-            print(f"\nIndividual results saved to {output} and {v_score_path}")
+           # v_score_path = output.replace("precision_at_10_statistics.csv", "v_score_statistics.csv")
+            #result_df_homogeneity_and_v_measure_score.to_csv(v_score_path, index=False)
+            #print(f"\nIndividual results saved to {output} and {v_score_path}")
             underperforming_population_frequency_path = output.replace("precision_at_10_statistics.csv", "err_population_frequency_statistics.csv")
             result_df_underperforming_population_frequency.to_csv(underperforming_population_frequency_path, index=False)
             print(f"\nIndividual results saved to {output} and {underperforming_population_frequency_path}")
@@ -355,11 +355,11 @@ def main(directory_path, output=None, aggregated_pop_ids=None):
 
 
 if __name__ == '__main__':
-    dir = ["domino_pca_weights:40_k=2", "kmeans_k=4", "spotlight_k=2", "george_k=3"]
-    base_directory_path = "./experiments/cbm/Waterbirds/Waterbirds_Attributes_with_Background/20251012-212000_Task_Baseline_wd:0.00001_lr:0.01_decrease:40_sgd/Evaluations_valEqTrain_v3/Slices/"
+    dir = ["domino_pca_weights:40_k=4", "kmeans_k=15", "spotlight_k=2", "george_k=4","hibug2_k=4"]
+    base_directory_path = "/homes/ea685/new_baseline/data/MNIST/MNIST-SUM/CBM_Joint/Evaluations/new_slices/"
     for d in dir:
         print(f"Calculating statistics for {d}")
         directory_path = os.path.join(base_directory_path, d)
         output = os.path.join(directory_path, "precision_at_10_statistics.csv")
-        aggregated_pop_ids = [[1, 2]]
+        aggregated_pop_ids = [[6, 12]]
         main(directory_path=directory_path, output=output, aggregated_pop_ids=aggregated_pop_ids)
diff --git a/baselines/run_silouette.py b/baselines/run_silouette.py
index d5a8934..e5134a4 100644
--- a/baselines/run_silouette.py
+++ b/baselines/run_silouette.py
@@ -7,13 +7,13 @@ from sklearn.cluster import KMeans
 from sklearn.metrics import silhouette_score
 from sklearn.mixture import GaussianMixture
 from domino import MixtureSlicer, SpotlightSlicer
-from utils import data_preprocessing, converter
+from utils.utils import data_preprocessing, converter  
 from umap import UMAP
 import pandas as pd
 
-algos = ["kmeans", "domino", "spotlight"]
+algos = ["kmeans", "domino", "spotlight", "george"]
 clusters = [2,4, 6, 8, 10, 12, 14, 16, 18, 20]
-df_path = "./experiments/cbm/Waterbirds/Waterbirds_Attributes_with_Background/20251012-212000_Task_Baseline_wd:0.00001_lr:0.01_decrease:40_sgd/Evaluations_valEqTrainV3/val_eval_df.pkl"
+df_path = "/homes/ea685/new_baseline/data/MNIST/MNIST-SUM/CBM_Sequential/Evaluations/val_eval_df.pkl"
 df = pd.read_pickle(df_path)
 
 data = data_preprocessing(df)
diff --git a/baselines/test_domino.py b/baselines/test_domino.py
index faebfeb..48dcd16 100644
--- a/baselines/test_domino.py
+++ b/baselines/test_domino.py
@@ -82,22 +82,22 @@ def main(val_data=None, config=None):
     stats_path = os.path.join(config["saving_path"], "metric_scores.csv")
     pd.DataFrame(stats_dict, index=[0]).to_csv(stats_path, index=False)
     
-    # Plot Only Erroneous Samples
-    plot_slices(
-        embeddings=test_embeddings[error_mask],
-        targets=test_labels[error_mask],
-        pred_labels=test_pred_labels[error_mask],
-        slices=slice_labels[error_mask],
-        saving_path=config["saving_path"],
-        dim_reduction_method='tsne',
-        fig_name='error_slices_tsne_plot_spotlight',
-        title="Predicted Error Slices Visualisation",
-    )
+    # # Plot Only Erroneous Samples
+    # plot_slices(
+    #     embeddings=test_embeddings[error_mask],
+    #     targets=test_labels[error_mask],
+    #     pred_labels=test_pred_labels[error_mask],
+    #     slices=slice_labels[error_mask],
+    #     saving_path=config["saving_path"],
+    #     dim_reduction_method='tsne',
+    #     fig_name='error_slices_tsne_plot_spotlight',
+    #     title="Predicted Error Slices Visualisation",
+    # )
     print("Done!")
 
 if __name__ == "__main__":
     
-    experiment_path = "./experiments/cbm/Waterbirds/20251204-110018_CBM_debugging_code_flow/"
+    experiment_path = "../../data/MNIST/MNIST-SUM/CBM_Joint/"
     evaluation_folder_name = "Evaluations"
     val_df_path = os.path.join(experiment_path, evaluation_folder_name, "val_eval_df.pkl")
 
@@ -132,9 +132,9 @@ if __name__ == "__main__":
     clip_image_embeddings = np.array(clip_image_embeddings_list)
     val_data["clip_image_embeddings"] = clip_image_embeddings
 
-    n_slices = [2]
+    n_slices = [4]
     for s in n_slices:
-        saving_path = os.path.join(*[experiment_path, evaluation_folder_name, "Slices", f"domino_pca_weights:40_k={s}"])
+        saving_path = os.path.join(*[experiment_path, evaluation_folder_name, "new_slices", f"domino_pca_weights:40_k={s}"])
         parent_path = os.path.dirname(saving_path)
         if not os.path.exists(parent_path):
             print(f"Creating saving directory: {parent_path}")
@@ -157,7 +157,7 @@ if __name__ == "__main__":
                 "saving_path": saving_path_seed,
                 'seed': seed,
                 'k': [5, 10],
-                'error_pop_inds': [1, 2],
+                'error_pop_inds': [6, 12],
                 'y_log_likelihood_weight': 40,
                 'y_hat_log_likelihood_weight': 40,
                 # [2, 6, 12],
diff --git a/baselines/test_george.py b/baselines/test_george.py
index e9bf197..bc7ff21 100644
--- a/baselines/test_george.py
+++ b/baselines/test_george.py
@@ -120,28 +120,28 @@ def main(val_data=None, test_data=None, config=None):
     pd.DataFrame(stats_dict, index=[0]).to_csv(stats_path, index=False)
     
     # Plot Only Erroneous Samples
-    plot_slices(
-        embeddings=test_data['embeddings'][error_mask],
-        targets=test_data['labels'][error_mask],
-        pred_labels=test_data['y_preds'][error_mask],
-        slices=slice_labels[error_mask],
-        saving_path=config["saving_path"],
-        dim_reduction_method='tsne',
-        fig_name='error_slices_tsne_plot_spotlight',
-        title="Predicted Error Slices Visualisation",
-    )
+    # plot_slices(
+    #     embeddings=test_data['embeddings'][error_mask],
+    #     targets=test_data['labels'][error_mask],
+    #     pred_labels=test_data['y_preds'][error_mask],
+    #     slices=slice_labels[error_mask],
+    #     saving_path=config["saving_path"],
+    #     dim_reduction_method='tsne',
+    #     fig_name='error_slices_tsne_plot_spotlight',
+    #     title="Predicted Error Slices Visualisation",
+    # )
     print("Done!")
 
 if __name__ == "__main__":
     
-    experiment_path = "./experiments/cbm/Waterbirds/20251204-110018_CBM_debugging_code_flow/"
+    experiment_path = "/homes/ea685/new_baseline/data/MNIST/MNIST-SUM/CBM_Joint/"
     evaluation_folder_name = "Evaluations"
     val_df_path = os.path.join(experiment_path, evaluation_folder_name, "val_eval_df.pkl")
     test_df_path = os.path.join(experiment_path, evaluation_folder_name, "val_eval_df.pkl")
 
-    n_slices = [3]
+    n_slices = [4]
     for s in n_slices:
-        saving_path = os.path.join(*[experiment_path, evaluation_folder_name, "Slices", f"george_k={s}"])
+        saving_path = os.path.join(*[experiment_path, evaluation_folder_name, "new_slices", f"george_k={s}"])
         parent_path = os.path.dirname(saving_path)
         if not os.path.exists(parent_path):
             print(f"Creating saving directory: {parent_path}")
@@ -163,7 +163,7 @@ if __name__ == "__main__":
                 "saving_path": saving_path_seed,
                 'seed': seed,
                 'k': [5, 10],
-                'error_pop_inds': [1, 2],
+                'error_pop_inds': [6, 12],
             }
 
             main(
diff --git a/baselines/test_k_means.py b/baselines/test_k_means.py
index 7542fec..54ea0c9 100644
--- a/baselines/test_k_means.py
+++ b/baselines/test_k_means.py
@@ -30,17 +30,17 @@ def main(val_data, test_data, config=None):
     # Plot error slices
     error_mask = test_pred_labels != test_targets
 
-    # Plot Only Erroneous Samples
-    plot_slices(
-        embeddings=test_embeddings[error_mask],
-        targets=test_targets[error_mask],
-        pred_labels=test_pred_labels[error_mask],
-        slices=slices[error_mask],
-        saving_path=config["saving_path"],
-        dim_reduction_method='tsne',
-        fig_name='error_slices_tsne_plot_kmeans',
-        title="Predicted Error Slices Visualisation",
-    )
+    # # Plot Only Erroneous Samples
+    # plot_slices(
+    #     embeddings=test_embeddings[error_mask],
+    #     targets=test_targets[error_mask],
+    #     pred_labels=test_pred_labels[error_mask],
+    #     slices=slices[error_mask],
+    #     saving_path=config["saving_path"],
+    #     dim_reduction_method='tsne',
+    #     fig_name='error_slices_tsne_plot_kmeans',
+    #     title="Predicted Error Slices Visualisation",
+    # )
 
     for k in config["k"]:
         precision_at_k_results = precision_at_k(
@@ -84,14 +84,14 @@ def main(val_data, test_data, config=None):
 
 
 if __name__ == "__main__":
-    experiment_path = "./experiments/cbm/Waterbirds/20251204-110018_CBM_debugging_code_flow/"
-    evaluation_folder_name = "Evaluations"
+    experiment_path = "../../data/MNIST/MNIST-SUM/CBM_Joint/"
+    evaluation_folder_name = "Evaluations/"
     val_df_path = os.path.join(experiment_path, evaluation_folder_name, "val_eval_df.pkl")
     test_df_path = os.path.join(experiment_path, evaluation_folder_name, "val_eval_df.pkl")
     
-    n_slices = [4]
+    n_slices = [15]
     for s in n_slices:
-        saving_path = os.path.join(*[experiment_path, evaluation_folder_name, "Slices", f"kmeans_k={s}"])
+        saving_path = os.path.join(*[experiment_path, evaluation_folder_name, "new_slices", f"kmeans_k={s}"])
         parent_path = os.path.dirname(saving_path)
         if not os.path.exists(parent_path):
             print(f"Creating saving directory: {parent_path}")
@@ -114,7 +114,7 @@ if __name__ == "__main__":
                 "saving_path": saving_path_seed,
                 'seed': seed,
                 "k": [5, 10],
-                "error_pop_inds": [1, 2],
+                "error_pop_inds": [6, 12],
             }
 
             main(
diff --git a/baselines/test_spotlight.py b/baselines/test_spotlight.py
index 784007b..3178646 100644
--- a/baselines/test_spotlight.py
+++ b/baselines/test_spotlight.py
@@ -92,21 +92,21 @@ def main(val_data=None, test_data=None, config=None):
     pd.DataFrame(stats_dict, index=[0]).to_csv(stats_path, index=False)
 
     # Plot Only Erroneous Samples
-    plot_slices(
-        embeddings=test_embeddings[error_mask],
-        targets=test_targets[error_mask],
-        pred_labels=test_pred_labels[error_mask],
-        slices=slice_labels[error_mask],
-        saving_path=config["saving_path"],
-        dim_reduction_method='tsne',
-        fig_name='error_slices_tsne_plot_spotlight',
-        title="Predicted Error Slices Visualisation",
-    )
+    # plot_slices(
+    #     embeddings=test_embeddings[error_mask],
+    #     targets=test_targets[error_mask],
+    #     pred_labels=test_pred_labels[error_mask],
+    #     slices=slice_labels[error_mask],
+    #     saving_path=config["saving_path"],
+    #     dim_reduction_method='tsne',
+    #     fig_name='error_slices_tsne_plot_spotlight',
+    #     title="Predicted Error Slices Visualisation",
+    # )
     print("Done!")
 
 if __name__ == "__main__":
     
-    experiment_path = "./experiments/cbm/Waterbirds/20251204-110018_CBM_debugging_code_flow/"
+    experiment_path = "../../data/MNIST/MNIST-SUM/CBM_Joint/"
     evaluation_folder_name = "Evaluations"
     val_df_path = os.path.join(experiment_path, evaluation_folder_name, "val_eval_df.pkl")
     test_df_path = os.path.join(experiment_path, evaluation_folder_name, "val_eval_df.pkl")
@@ -115,7 +115,7 @@ if __name__ == "__main__":
     test_df = pd.read_pickle(test_df_path)
 
     for s in [2]:
-        saving_path = os.path.join(*[experiment_path, evaluation_folder_name, "Slices", f"spotlight_k={s}"])
+        saving_path = os.path.join(*[experiment_path, evaluation_folder_name, "new_slices", f"spotlight_k={s}"])
         parent_path = os.path.dirname(saving_path)
         if not os.path.exists(parent_path):
             print(f"Creating saving directory: {parent_path}")
@@ -135,7 +135,7 @@ if __name__ == "__main__":
                 "saving_path": s_saving_path,
                 'k': [5, 10],
                 "seed": 42,
-                "error_pop_inds": [1, 2],
+                "error_pop_inds": [6, 12],
                 "spotlight_size": seed,
             }
 
